docs/architecture/adr-016-reactions-module.md

```diff
@@ -0,0 +1,338 @@
+  
+  // RemoveRegisteredReaction allows to remove an existing supported reaction
+  rpc RemoveRegisteredReaction(MsgRemoveRegisteredReaction) returns (MsgRemoveRegisteredReactionResponse);
+  
+  // SetReactionsParams allows to set the reactions params
+  rpc SetReactionsParams(MsgSetReactionsParams) returns (MsgSetReactionsParamsResponse);
+}
+
+// MsgAddReaction represents the message to be used to add a post reaction
+message MsgAddReaction {
+  // Id of the subspace inside which the post to react to is 
+  uint64 subspace_id = 1;
+  
+  // Id of the post to react to
+  uint64 post_id = 2;
+  
+  // Value of the reaction
+  google.proto.Any value = 3;
+  
+  // User reacting to the post
+  string user = 4;
+}
+
+// MsgAddReactionResponse represents the Msg/AddReaction response type
+message MsgAddReactionResponse {
+  // Id of the newly added reaction
+  uint64 reaction_id = 1;
+}
+
+// MsgRemoveReaction represents the message to be used to remove an 
+// existing reaction from a post
+message MsgRemoveReaction {
+  // Id of the subspace inside which the reaction to remove is
+  uint64 subspace_id = 1;
+  
+  // Id of the reaction to be removed
+  uint64 reaction_id = 2;
+  
+  // User removing the reaction
+  string user = 3;
+}
+
+// MsgRemoveReactionResponse represents the Msg/RemoveReaction response type
+message MsgRemoveReactionResponse {}
+
+// MsgAddRegisteredReaction represents the message to be used to 
+// register a new supported reaction
+message MsgAddRegisteredReaction {
+  // Id of the subspace inside which this reaction should be registered
+  uint64 subspace_id = 1;
+  
+  // Shorthand code of the reaction
+  string shorthand_code = 2;
+  
+  // Display value of the reaction
+  string display_value = 3;
+  
+  // User adding the supported reaction
+  string user = 4;
+}
+
+// MsgAddRegisteredReactionResponse represents the 
+// Msg/AddRegisteredReaction response type
+message MsgAddRegisteredReactionResponse {
+  // If of the newly registered reaction
+  uint32 registered_reaction_id = 1;
+}
+
+// MsgRemoveRegisteredReaction represents the message to be used to
+// remove an existing registered reaction
+message MsgRemoveRegisteredReaction {
+  // Id of the subspace from which to remove the registered reaction
+  uint64 subspace_id = 1;
+  
+  // Id of the registered reaction to be removed
+  uint32 registered_reaction_id = 2;
+  
+  // User removing the registered reaction
+  string user = 3;
+}
+
+// MsgRemoveRegisteredReactionResponse represents the 
+// Msg/RemoveRegisteredReaction response type
+message MsgRemoveRegisteredReactionResponse {}
+
+// MsgSetReactionsParams represents the message to be used when setting 
+// a subspace reactions params
+message MsgSetReactionsParams {
+  // Id of the subspace for which to set the params
+  uint64 subspace_id = 1;
+
+  // Params related to RegisteredReactionValue reactions
+  RegisteredReactionValueParams registered = 2;
```

Comment:
I think you miswrote here
```suggestion
  RegisteredReactionValueParams reactions = 2;
```

Reply:
I've updated the name to make it clearer